source/app-development/tutorials-interactive-apps/k8s-like-hpc-jupyter.rst

```diff
@@ -0,0 +1,371 @@
+.. _app-development-tutorials-interactive-apps-k8s-like-hpc-jupyter:
+
+Add a Jupyter App on a Kubernetes Cluster that behaves like HPC compute
+=======================================================================
+
+This tutorial will walk you through creating an interactive Jupyter app that
+your users will use to launch a `Jupyter Notebook Server`_ in a kubernetes cluster.
+The container will behave much like a HPC compute node. This has the benefit that
+a single app can serve both traditional HPC as well as Kubernetes.
+
+It assumes you have a working understanding of app development already. The purpose of
+this document is to describe how to write apps specifically for a kubernetes cluster,
+so it skips a lot of important details about app development that may be found in
+other tutorials like :ref:`app-development-tutorials-interactive-apps-add-jupyter`.
+
+
+We're going to be looking at the `bc osc jupyter`_ app which you can fork, clone
+and modify for your site.  This page also holds the `submit yml in full`_ for reference.
+
+The container
+-------------
+
+The container to make Kubernetes pods look like HPC compute would end up needing all
+the OS packages present on the HPC environment.  The OS of the container itself would also
+need to compatible with packages installed on HPC environment so if you run RHEL 8 on HPC
+you would also need to run RHEL 8 inside the container.
+
+Things like Lmod and HPC applications will need to be run inside the Pod's container just like
+if a job was spawned in the traditional HPC resource manager.
+
+
+Switch between SLURM and Kubernetes
+-----------------------------------
+
+The first big change from a traditional HPC interactive app is the main YAML structure is wrapped
+in a large ``if`` statement based on the cluster choice. If a user choose one of the HPC clusters,
+the SLURM submit YAML is rendered, otherwise the Kubernetes YAML is rendered.
```

Comment:
```suggestion
in a large ``if`` statement based on the cluster choice. If a user chooses one of the SLURM clusters,
the SLURM submit YAML is rendered, otherwise the Kubernetes YAML is rendered.
```

Reply:
Actually this sentence should be reworded to describe our environment. Notably that Owens and Pitzer are Slurm and that that our kubernetes cluster is just called 'kubernetes'.

Reply:
Tried to reword describing environment